### ISSUE TITLE
Have ipcluster pass --location arg to ipcontroller

### ISSUE THREAD

**Issue by dereneaton** [ipython/ipyparallel#231]
tldr;
It seems like `--location=` would be a convenient argument for `ipcluster` to ensure that engines can find the controller on a multiple machine setup when using `--ip=*`. 

As you say in the docs (http://ipyparallel.readthedocs.io/en/latest/process.html#general-considerations) the `--ip=*` argument to `ipcluster` will listen on all interfaces to find the engines on a secure network (like a University HPC), but the engines will not always know the location of the controller, though they try to guess. If ipcluster accepted the `--location` argument and used it to start the `ipcontroller` instance then it seems like the engines would be able to find it, unless perhaps the details are more complicated under the hood. 

My thought was that I could start a cluster across 3 20-core nodes on multiple machines by running the following:
```
ipcluster start --n=60 --engines=MPI --ip=* --location=$(hostname -i)
```

When the nodes are on the same physical machine it finds all 60 cores just fine (without a --location arg), but when they are on different machines I can only find them all by starting ipcontroller and ipengine steps separately.

**Comment by minrk** [comment]
Good idea. #233 should add this, and improve the defaults, so that `--location=$(hostname)` is the default.
